base_location/tests/test_base_location.py

```diff
@@ -0,0 +1,315 @@
+# Copyright 2015 Yannick Vaucher, Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+from odoo.exceptions import ValidationError
+from odoo.tests import tagged
+from odoo.tools.misc import mute_logger
+import psycopg2
+
+
+@tagged('post_install', '-at_install')
+class TestBaseLocation(TransactionCase):
+
+    def setUp(self):
+        super(TestBaseLocation, self).setUp()
+        self.state_vd = self.env['res.country.state'].create({
+            'name': 'Vaud',
+            'code': 'VD',
+            'country_id': self.ref('base.ch'),
+        })
+        self.env['res.country'].browse(self.ref('base.es')).write({
+            'enforce_cities': True
+        })
+        self.company = self.env.ref('base.main_company')
+
+        self.state_bcn = self.env['res.country.state'].create({
+            'name': 'Barcelona',
+            'code': '08',
+            'country_id': self.ref('base.es'),
+        })
+        self.state_madrid = self.env['res.country.state'].create({
+            'name': 'Madrid',
+            'code': '28',
+            'country_id': self.ref('base.es'),
+        })
+        self.city_bcn = self.env['res.city'].create({
+            'name': 'Barcelona',
+            'state_id': self.state_bcn.id,
+            'country_id': self.ref('base.es'),
+        })
+        self.city_madrid = self.env['res.city'].create({
+            'name': 'Madrid',
+            'state_id': self.state_madrid.id,
+            'country_id': self.ref('base.es'),
+        })
+        self.city_lausanne = self.env['res.city'].create({
+            'name': 'Lausanne',
+            'state_id': self.state_vd.id,
+            'country_id': self.ref('base.ch'),
+        })
+
+    def values_city_zip1(self):
+        return {
+            'name': 1000,
+            'city_id': self.city_lausanne.id,
+        }
+
+    def values_city_zip2(self):
+        return {
+            'name': 2000,
+            'city_id': self.city_bcn.id,
+        }
+
+    def test_onchange_partner_city_completion(self):
+        """Test whether all data in partner is filled correctly
+        after changing it's zip"""
+        partner1 = self.env['res.partner'].new({
+            'name': 'Camptocamp',
+        })
+        city_zip2 = self.env['res.city.zip'].create(
+            self.values_city_zip2())
+        city_zip2.city_id.country_id.enforce_cities = True
+        partner1.zip_id = city_zip2
+        partner1._onchange_zip_id()
+        self.assertEqual(partner1.zip, city_zip2.name)
+        self.assertEqual(partner1.city, city_zip2.city_id.name)
+        self.assertEqual(partner1.state_id, city_zip2.city_id.state_id)
+        self.assertEqual(partner1.country_id, city_zip2.city_id.country_id)
+
+    def test_onchange_company_city_completion(self):
+        """Test whether all data in company is filled correctly
```

Comment:
same, you can shorten this docstrings and make them more PEP compliant (not mandatory as we don't enforce this yet in OCA modules)